Storage/ThirdParty/SwiftData.swift

```diff
@@ -104,7 +104,7 @@ public class SwiftData {
      * close a database connection and run a block of code inside it.
      */
     func withConnection(flags: SwiftData.Flags, synchronous: Bool=true, cb: (db: SQLiteDBConnection) -> NSError?) -> NSError? {
-        let conn: ConcreteSQLiteDBConnection?
+        weak var conn: ConcreteSQLiteDBConnection?
```

Comment:
Note that you won't need to change the signature of `cb` to be `SQLiteDBConnection?`, because although weak references must be optionally typed, it's our dispatch blocks that will handle that and create a `FailedSQLiteDBConnection` to pass back to `cb`.